iosxr_test.py

```diff
@@ -0,0 +1,262 @@
+            print('Timed out - nameserver 208.67.220.220 is not populated in resolv.conf')
+            return False
+        s.prompt()
+
+        s.sendline("cat /etc/resolv.conf | grep 222")
+        result = s.expect(['nameserver 208.67.222.222', pexpect.EOF, pexpect.TIMEOUT])
+        if result == 0:
+            print('==>nameserver 208.67.222.222 is successfully populated')
+        elif result == 1:
+            print('EOF')
+            return False
+        elif result == 2:
+            print('Timed out - nameserver 208.67.220.222 is not populated in resolv.conf')
+            return False
+        s.prompt()
+
+        print('Check vagrant public key has been replaced by private:')
+        s.sendline('grep "public" ~/.ssh/authorized_keys -c')
+        result = s.expect(['0', pexpect.EOF, pexpect.TIMEOUT])
+        if result == 0:
+            print('==>SSH public key successfully replaced')
+        elif result == 1:
+            print('EOF')
+            return False
+        elif result == 2:
+            print('Timed out - SSH public key not successfully replaced')
+            return False
+
+        s.prompt()
+        s.logout()
+    except pxssh.ExceptionPxssh as e:
+        print("pxssh failed on login.")
+        print(e)
+        return False
+    else:
+        print("Vagrant SSH to XR Linux is sane")
+        return True
+
+
+def test_xr():
+    '''
+    Log into IOS XR Console and run some basic sanity tests.
+
+    Verify logging into IOS XR Console directly.
+    Verify show version.
+    Verify show run.
+    '''
+
+    print('Testing XR Console...')
+
+    try:
+        s = pxssh.pxssh()
+        s.force_password = True
+        s.PROMPT = 'RP/0/RP0/CPU0:ios# '
+
+        print('Check logging into XR Console')
+        s.login(hostname, username, password, terminal_type, xr_prompt, login_timeout, xr_port, auto_prompt_reset=False)
+        s.prompt()
+        s.sendline('term length 0')
+        s.prompt()
+        print('==>Successfully logged into XR Console')
+
+        print('Check show version')
+        s.sendline('show version | i cisco IOS XRv x64')
+        result = s.expect(['XRv x64', pexpect.EOF, pexpect.TIMEOUT])
+        if result == 0:
+            print('==>XRv x64 correctly found in show version')
+        elif result == 1:
+            print('EOF')
+            return False
+        elif result == 2:
+            print('Timed out - show version failed')
+            return False
+        s.prompt()
+
+        print('Check show run grpc')
+        s.sendline('show run grpc')
+        result = s.expect(['port 57777', pexpect.EOF, pexpect.TIMEOUT])
+        if result == 0:
+            print('==>GRPC port 57777 correctly found in show run')
+        elif result == 1:
+            print('EOF')
+            return False
+        elif result == 2:
+            print('Timed out - show run failed')
+            return False
+
+        s.prompt()
+        s.logout()
+
+    except pxssh.ExceptionPxssh as e:
```

Comment:
Does all of the above need to be in this `try` block, or only up to the "Successfully logged in" part?